front/app/component-library/components/CollapsibleContainer/index.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { ReactNode, useState } from 'react';
+
+import { CSSTransition } from 'react-transition-group';
+import styled from 'styled-components';
+
+import useInstanceId from '../../hooks/useInstanceId';
+import { isRtl } from '../../utils/styleUtils';
+import Box, { BoxMarginProps, BoxPaddingProps, BoxWidthProps } from '../Box';
+import Icon from '../Icon';
+import Title, { Variant, FontSize, FontWeight } from '../Title';
+
+type CollapsibleContainerProps = {
+  children: ReactNode;
+  title: ReactNode;
+  isOpenByDefault?: boolean;
+  useRegionRole?: boolean;
+} & BoxMarginProps &
+  BoxWidthProps &
+  BoxPaddingProps &
+  TitleProps;
+
+type TitleProps = {
+  titleVariant?: Variant;
+  titleFontSize?: FontSize;
+  titlePadding?: string;
+  titleFontWeight?: FontWeight;
+};
```

Comment:
Since we wrap the TitleButton in a Title (Header) element now, we need to be able to pass in styling options for the Title (since it's not correct html to have a Header in a Header). I had to add a few props for the title styling options we need at the moment.